src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/EntityComponent.scala

```diff
@@ -42,6 +43,33 @@ trait EntityComponent {
     type EntityQuery = Query[EntityTable, EntityRecord, Seq]
     type EntityQueryWithAttributesAndRefs =  Query[(EntityTable, Rep[Option[(AttributeTable, Rep[Option[EntityTable]])]]), (EntityRecord, Option[(AttributeRecord, Option[EntityRecord])]), Seq]
 
+    // result structure from entity and attribute list raw sql
+    case class EntityListResult( workspaceId: UUID,
+                                 entityName: String,
+                                 entityType: String,
+                                 entityId: Long,
+                                 attributeId: Option[Long], // if this is None it means there are no attributes
+                                 listIndex: Option[Int],
+                                 valueString: Option[String],
+                                 valueNumber: Option[Double],
+                                 valueBoolean: Option[Boolean],
+                                 attributeName: Option[String], // if this is None it means there are no attributes
+                                 refId: Option[Long], // if any of the ref fields are None it means it is not a reference
+                                 refName: Option[String],
+                                 refType: Option[String])
```

Comment:
for future work (don't hold the PR for this), it may be better to do something like:

```
case class EntityListResult(
  entityRecord: EntityRecord,
  attributeRecord: Option[AttributeRecord],
  refEntityRecord: Option[EntityRecord]
)
```

above code written in a PR vacuum, may not be correct, but you get the idea.